src/world_new/WorldData.cpp

```diff
@@ -9,58 +9,79 @@
+        }
+
+
+        setViewVectors();
     }
 
-    /*
-     * Reserve the us and them vectors, making sure they are not being resized (which will invalidate the references)
-     */
-    auto amountUs = ours.size();
-    auto amountThem = others.size();
-    robots.reserve(amountUs + amountThem);
-    us.reserve(amountUs);
-    them.reserve(amountThem);
-    robotsNonOwning.reserve(amountUs + amountThem);
-
-    for (auto &each : ours) {
-        auto _ptr = &robots.emplace_back(feedback, each, Team::us, getBall());
-        us.emplace_back(_ptr);
-        robotsNonOwning.emplace_back(_ptr);
+    std::vector<view::RobotView> const &WorldData::getUs() const noexcept {
+        return us;
     }
-    for (auto &each : others) {
-        auto _ptr = &robots.emplace_back(feedback, each, Team::them, getBall());
-        them.emplace_back(_ptr);
-        robotsNonOwning.emplace_back(_ptr);
+
+    std::vector<view::RobotView> const &WorldData::getThem() const noexcept { return them; }
+
+    std::vector<rtt::world_new::robot::Robot> const &WorldData::getRobots() const noexcept { return robots; }
+
+    std::optional<view::BallView> WorldData::getBall() const noexcept {
+        if (ball.has_value()) {
+            return std::optional<view::BallView>(&ball.value());
+        } else {
+            return std::nullopt;
+        }
     }
-}
 
-std::vector<view::RobotView> const &WorldData::getUs() const noexcept {
-    return us;
-}
+    bool WorldData::weHaveRobots() const noexcept { return !getUs().empty(); }
 
-std::vector<view::RobotView> const &WorldData::getThem() const noexcept { return them; }
+    uint64_t WorldData::getTime() const noexcept { return time; }
 
-std::vector<rtt::world_new::robot::Robot> const &WorldData::getRobots() const noexcept { return robots; }
+    std::vector<view::RobotView> const &WorldData::getRobotsNonOwning() const noexcept { return robotsNonOwning; }
 
-std::optional<view::BallView> WorldData::getBall() const noexcept {
-    if (ball.has_value()) {
-        return std::optional<view::BallView>(&ball.value());
-    } else {
-        return std::nullopt;
-    }
-}
+    WorldData &WorldData::operator=(WorldData const & other) noexcept {
+        if (this == &other) {
+            return *this;
+        }
+
+        this->robots = other.robots;
+        this->ball = other.ball;
+
+        setViewVectors();
 
-bool WorldData::weHaveRobots() const noexcept { return !getUs().empty(); }
+        return *this;
+    }
 
-uint64_t WorldData::getTime() const noexcept { return time; }
+    WorldData::WorldData(WorldData const & other) noexcept :
+            robots{ other.robots }, ball{ other.ball }{
+        setViewVectors();
+    }
 
-std::vector<view::RobotView> const &WorldData::getRobotsNonOwning() const noexcept { return robotsNonOwning; }
+    void WorldData::setViewVectors() noexcept {
+        /*
+         * Reserve the us and them vectors, making sure they are not being resized (which will invalidate the references)
```

Comment:
where does this reserve happen?

Reply:
Oh yeah, nice catch, ill take a look at that tomorrow